akka-cluster-tools/src/main/scala/akka/cluster/pubsub/DistributedPubSubMediator.scala

```diff
@@ -615,15 +619,21 @@ class DistributedPubSubMediator(settings: DistributedPubSubSettings) extends Act
     case msg @ Unsubscribed(ack, ref) ⇒
       ref ! ack
 
-    case Status(otherVersions) ⇒
-      // gossip chat starts with a Status message, containing the bucket versions of the other node
-      val delta = collectDelta(otherVersions)
-      if (delta.nonEmpty)
-        sender() ! Delta(delta)
-      if (otherHasNewerVersions(otherVersions))
-        sender() ! Status(versions = myVersions) // it will reply with Delta
+    case Status(otherVersions, reply) ⇒
+      // only accept status from known nodes, otherwise old cluster with same address may interact
+      // also accept from local for testing purposes
+      if (nodes(sender().path.address) || sender().path.address.hasLocalScope) {
+        // gossip chat starts with a Status message, containing the bucket versions of the other node
+        val delta = collectDelta(otherVersions)
+        if (delta.nonEmpty)
+          sender() ! Delta(delta)
+        if (!reply && otherHasNewerVersions(otherVersions))
```

Comment:
`reply` is probably a misnomer as some kind of reply is clearly sent. Can you replace it with a more descriptive name?